kgtk/gt/embedding_utils.py

```diff
@@ -650,38 +690,47 @@ def dump_vectors(self, file_name, type_=None):
                     for i in each:
                         _ = f.write(str(i) + "\t")
                     _ = f.write("\n")
+    def print_vector_line(self, vector_line, output_file):
+        if output_file == "":
+            print(vector_line,end='')
+        else:
+            output_file.write(vector_line)
 
     def print_vector(self, vectors, output_properties: str = "text_embedding",
-                     output_format="kgtk_format", save_embedding_sentence=False):
+                     output_format="kgtk_format", save_embedding_sentence=False, output_file=""):
         self._logger.debug("START printing the vectors")
+        if output_file != "":
+            output_file = open(output_file, "w")
         if output_format == "kgtk_format":
             # TODO: This should be comverted to use KgtkWriter
-            print("node\tproperty\tvalue\n", end="")
+            self.print_vector_line("node\tproperty\tvalue\n", output_file)
```

Comment:
is this line writing to the output file? I suggest using `KGTKWriter` to do so. Also please change the code to use `KGTKReader` to read in the file.

You can find how-to in `kgtk cat`, for example